source/common/http/conn_manager_impl.cc

```diff
@@ -721,11 +721,13 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(ActiveStreamDecoderFilte
     ASSERT(!(state_.filter_call_state_ & FilterCallState::DecodeHeaders));
     state_.filter_call_state_ |= FilterCallState::DecodeHeaders;
     FilterHeadersStatus status = (*entry)->decodeHeaders(
-        headers, end_stream && continue_data_entry == decoder_filters_.end());
+        headers,
+        decoding_headers_only_ || (end_stream && continue_data_entry == decoder_filters_.end()));
```

Comment:
Do we want to allow `decodeHeaders(headers, end_stream=true)` to return `ContinueHeadersOnly`? I don't think we should, so add an ASSERT?

Reply:
+1 on ASSERTing this.